### PR TITLE
README.md for RSA Algorithm

### LINKED ISSUE #2162: RSA algorithm
- [x] C
- [x] C++
- [ ] C#
- [ ] CoffeeScript
- [ ] Dart
- [ ] Go - @KamalDGRT
- [x] Java
- [ ] JS - @MOHITSAHAL
- [ ] Kotlin - @RaghavAwasthi
- [ ] Php
- [x] Python
- [ ] Readme - @mharshita
- [ ] Ruby
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by mharshita**
Fixes #2162 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added README.md 

#### Languages Used:
- Mardown

#### Files Added:
- README.md

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by mharshita** [comment]
@jainaman224  @somya-kapoor  Please review my PR.

File: RSA_Algorithm/README.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# **RSA Algorithm**
+### RSA (Rivest-Shamir-Adleman) is an algorithm used by modern computers to encrypt and decrypt messages. It is an asymmetric cryptographic algorithm. Asymmetric means that there are two different keys. This is also called public key cryptoraphy, because one of the keys can be given to anyone. The other key must be kept private. The algorithm is based on the fact that finding the factors of large composite number is difficult: when the factors are prime numbers, the problem is called prime factorization. It is also a key pair (public and private key) generator.
```
Comment (asha15) [inline RSA_Algorithm/README.md:2]: please remove the '#' sign.

File: RSA_Algorithm/README.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+![rsa algorithm](https://www.educative.io/api/edpresso/shot/5284561120395264/image/5318859621924864)
+
+## **Algorithm**
+---
+- ### **Generating the keys**
+  ### 1.Generate two large random primes, p and q, of approximately equal size such that their product n=pq is of the required bit length.
+  ### 2.Compute n=pq and ϕ=(p−1)(q−1).
+  ### 3.Choose an integer e, 1<e<ϕ, such that gcd(e,ϕ)=1.
+  ### 4.Compute the secret exponent d, 1<d<ϕ, such that ed≡1modϕ. 
+  ### 5.The public key is (n,e) and the private key (d,p,q). Keep all the values d, p, q and ϕ secret. [Sometimes the private key is written as (n,d) because you need the value of n when using d. Other times we might write the key pair as ((N,e),d).]
```
Comment (asha15) [inline RSA_Algorithm/README.md:13]: please remove the '#' sign.

File: RSA_Algorithm/README.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+## **Encryption**
+---
+### Given a plaintext P, represented as a number, the ciphertext C is calculated as:
+### C = P^{e} mod n
```
Comment (asha15) [inline RSA_Algorithm/README.md:18]: please remove the '#' sign.

File: RSA_Algorithm/README.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+## **Decryption**
+---
+### Using the private key (n,d)(n,d), the plaintext can be found using:
+### P=C^{d} mod n
```
Comment (asha15) [inline RSA_Algorithm/README.md:23]: please remove the '#' sign.

File: RSA_Algorithm/README.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
```
Comment (asha15) [inline RSA_Algorithm/README.md:24]: remove the extra line

File: RSA_Algorithm/README.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+![encryption](https://miro.medium.com/max/1172/1*bEjF4alLo2CsazW-xWaxug.png)
+
+## **Pseudocode**
+---
```
Comment (asha15) [inline RSA_Algorithm/README.md:29]: please include the code inside ```

File: RSA_Algorithm/README.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+## **Complexity**
+---
+### Three  major components of the RSA algorithm are exponentiation, inversion and modular operation. Time complexity of the algorithm heavily depends on the complexity of the sub modules used.  We can take the liberty to  perform modular addition in cryptography in O(logn) where n is the number of digits in any number in Z.
+
+### Now modular multiplication using squaring and multiply technique  to get memodNcan be  shown as multiplying the digits of m log e times.  Considering the complexity of multiplication O({logn}2) i.e. repeated addition of two number of logn bits each,  the complexity of the modular exponentiation is about  O({logn}3)
+
+### Using Euclidean extended GCD, inverse of a number can be calculated in O({logn}2)
+
+### Thus overall time complexity of the key generation algorithm will be O(N2) encryption and decryption algorithm using squaring and multiply technique will be O(N3):For N digit number space.
```
Comment (asha15) [inline RSA_Algorithm/README.md:82]: please remove the '#' sign.

File: RSA_Algorithm/README.md (hunk 1 above)
Comment (mharshita) [reply RSA_Algorithm/README.md]: @asha15   Required changes have been done.

File: RSA_Algorithm/README.md (hunk 2 above)
Comment (mharshita) [reply RSA_Algorithm/README.md]: @asha15  Required changes have been done.

File: RSA_Algorithm/README.md (hunk 3 above)
Comment (mharshita) [reply RSA_Algorithm/README.md]: @asha15  Required changes have been done.

File: RSA_Algorithm/README.md (hunk 4 above)
Comment (mharshita) [reply RSA_Algorithm/README.md]: @asha15  Required changes have been done.

File: RSA_Algorithm/README.md (hunk 5 above)
Comment (mharshita) [reply RSA_Algorithm/README.md]: @asha15  Required changes have been done.

File: RSA_Algorithm/README.md (hunk 6 above)
Comment (mharshita) [reply RSA_Algorithm/README.md:27]: @asha15  Required changes have been done.

File: RSA_Algorithm/README.md (hunk 7 above)
Comment (mharshita) [reply RSA_Algorithm/README.md]: @asha15  Required changes have been done.

**Comment by mharshita** [comment]
@jainaman224  @asha15  @somya-kapoor  Please review my PR. The required changes have been done.

File: RSA_Algorithm/README.md [hunk 8]
```diff
@@ -0,0 +1,85 @@
+**RSA Algorithm**
```
Comment (asha15) [inline RSA_Algorithm/README.md:1]: Don't bold it, make this as a H1 tag by using #

Comment (mharshita) [reply RSA_Algorithm/README.md]: @asha15 Required changes have been made.

**Comment by mharshita** [comment]
@asha15 @jainaman224  I have made the required changes and squashed the commits. Please review.

**Pull request merged**